### PR TITLE
:tada: Source Amplitude: enable data region selection

### PR DESCRIPTION
**Pull request by mariamthiam**
## What
Source Amplitude, enable data region selection describe in issue https://github.com/airbytehq/airbyte/issues/19605

## How
Add parameter `data_region` to make possible to select data server region 

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
No breaking changes,  default value of region keep the same working method .

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py [hunk 1]
```diff
@@ -21,10 +21,17 @@
...
+    url_base = ""
     api_version = 2
 
+    def __init__(self, data_region: str, **kwargs):
+        super().__init__(**kwargs)
+        self.url_base = (
+            "https://analytics.eu.amplitude.com/api/"
+            if data_region == "EU Residency Server"
+            else "https://amplitude.com/api/"
+        )
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:33]: Would be better transform the `url_base` into a property function @mariamthiam. Check how Zendeks Sunshine does it: https://github.com/airbytehq/airbyte/blob/f11507b23a67f591a3c37abd8169d68c9aa1f1df/airbyte-integrations/connectors/source-zendesk-sunshine/source_zendesk_sunshine/streams.py#L26-L28

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:33]: Do you mind adding a link to Amplitude docs explaining about regions?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments.

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py (hunk 1 above)
Comment (mariamthiam) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:33]: Yes here is [API documentation](https://www.docs.developers.amplitude.com/analytics/apis/chart-annotations-api/#authorization) of Amplitude  , in the `Endpoints` section you have the different server.

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py [hunk 2]
```diff
@@ -21,10 +21,17 @@
...
-    url_base = "https://amplitude.com/api/"
     api_version = 2
 
+    def __init__(self, data_region: str, **kwargs):
+        self.data_region = data_region
+        super().__init__(**kwargs)
+
+    @property
+    def url_base(self) -> str:
+        subdomain = "analytics.eu." if self.data_region == "EU Residency Server" else ""
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:32]: Are there more than EU + US?

**Review by bazarnov — APPROVED** [review]
There is room for improvement probably we can improve passing input params and limit them to just `config` and pass only the `config` to each stream class and parse it internally to look for different parts. However, this one looks good. LGTM!

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py (hunk 2 above)
Comment (mariamthiam) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:32]: No according to the documentation https://www.docs.developers.amplitude.com/analytics/apis/http-v2-api/

**Review by marcosmarxm — APPROVED** [review]

**Comment by pedroslopez** [comment]
Hey @mariamthiam @marcosmarxm - this PR introduced a breaking change that affected existing syncs. It looks like our test suite caught this breaking change, but it was muted. 

Even though the required field has a default, the default will not be set for configs that were created beforehand. This is leading to failed syncs upon upgrading the connector:

```
Config validation error: 'data_region' is a required property

Traceback (most recent call last):
  File "/airbyte/integration_code/main.py", line 13, in <module>
    launch(source, sys.argv[1:])
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 131, in launch
    for message in source_entrypoint.run(parsed_args):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 104, in run
    raise traced_exc
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 98, in run
    check_config_against_spec_or_exit(connector_config, source_spec)
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/utils/schema_helpers.py", line 161, in check_config_against_spec_or_exit
    raise AirbyteTracedException(
airbyte_cdk.utils.traced_exception.AirbyteTracedException: 'data_region' is a required property
```

Could you please change this field to not be required and handle this being empty on the connector side? Thanks!

**Pull request merged**